### ISSUE TITLE
Invalidating method usage refcount(?) from a plugin

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#1879]
In `psalm/phpunit-psalm-plugin` I'm marking dataProviders as [being used by test methods](https://github.com/psalm/phpunit-psalm-plugin/blob/d2e2c43aa15c6c7ded9140ab76ca128a71a0ed28/hooks/TestCaseHandler.php#L171-L174):

```php
// methodExists also can mark methods as used (weird, but handy)
if (null === $provider_method_id
    || !$codebase->methodExists($provider_method_id, $provider_docblock_location, $declaring_method_id)
) {
```
However this has recently became unreliable, under the following scenario:
1. Create a test case with a dataProvider, reference dataProvider in a test method (`@dataProvider` tag)
2. Run Psalm with PhpUnit plugin enabled
3. Remove the `@dataProvider` phpdoc tag
4. Run Psalm (with plugin enabled) with `--find-dead-code`
5. Psalm thinks the data provider is still used.

Is there a way to selectively remove all the usage data for a given method? Something like this:
```php
foreach ($codebase->findAllUsedSymbolsBy($method_id) as $symbol) {
      // assuming data provider is marked as such when reference is added
      if ($symbol->custom_metadata[__NAMESPACE__]['is_data_provider'] ?? false) {
          $codebase->methodDoesNoLongerUse($method_id, $symbol);
      }
}
```

**Comment by muglug** [comment]
> Is there a way to selectively remove all the usage data for a given method? Something like this

Not at the moment - any changes detected at the parsing step [are used to invalidate those records](https://github.com/vimeo/psalm/blob/8aadf93402d84d95f448ba5dc49c4994fa1760bc/src/Psalm/Internal/Codebase/Analyzer.php#L513-L766).

But that logic should only be necessary when running with `--diff-methods` - a full Psalm run _should_ reset all that data. If that's not happening, something's amiss.

**Comment by weirdan** [comment]
Ok, this appear to have been fixed in 3.2.11, so I'm just gonna bump the version constraint instead.
